pkg/nsx/services/vpc/vpc.go

```diff
@@ -471,3 +523,236 @@ func (s *VPCService) Cleanup() error {
 
 	return nil
 }
+
+func (service *VPCService) needUpdateRule(rule *model.Rule, externalCIDR []string) bool {
+	des := rule.DestinationGroups
+	currentDesSet := sets.Set[string]{}
+	for _, group := range des {
+		currentDesSet.Insert(group)
+	}
+	if len(externalCIDR) != len(currentDesSet) {
+		return true
+	}
+	for _, newGroup := range externalCIDR {
+		if !currentDesSet.Has(newGroup) {
+			return true
+		}
+	}
+	return false
+}
+
+func (service *VPCService) getExternalIpBlock(vpcName string, namespace string) ([]string, error) {
+	// read corresponding vpc network config from store
+	nc_name, err := service.getNetworkconfigNameFromNS(namespace)
+	if err != nil {
+		log.Error(err, "failed to get network config name for VPC when creating NSX VPC rule", "VPC", vpcName, "namespace", namespace)
+		return nil, err
+	}
+	nc, _exist := service.GetVPCNetworkConfig(nc_name)
+	if !_exist {
+		log.Error(err, "failed to read network config  when creating NSX VPC rule", "nc_name", nc_name)
```

Comment:
when creating NSX VPC rule? it looks like we are doing to get exertnal ip block instead of creating nsx vpc rule?

Reply:
once vpc created. we need the external ip block cidr.